apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs

```diff
@@ -0,0 +1,175 @@
+        [Fact]
+        public async Task TestBindStringNull() =>
+            await TestBind<double?>(SpannerDbType.String, null);
+        [Fact]
+        public async Task TestBindStringArray() =>
+            await TestBind(SpannerDbType.ArrayOf(SpannerDbType.String), new[]
+                { "abc", null, "123"});
+        [Fact]
+        public async Task TestBindStringEmptyArray() =>
+            await TestBind(SpannerDbType.ArrayOf(SpannerDbType.String), new string[] {});
+        [Fact]
+        public async Task TestBindStringNullArray() =>
+            await TestBind<string[]>(SpannerDbType.ArrayOf(SpannerDbType.String), null);
+
+        [Fact]
+        public async Task TestBindByteArray() =>
+            await TestBind(SpannerDbType.Bytes, new byte[] { 1, 2, 3 });
+        [Fact]
+        public async Task TestBindByteArrayNull() =>
+            await TestBind<double?>(SpannerDbType.Bytes, null);
+
+        [Fact]
+        public async Task TestBindDate() =>
+            await TestBind(SpannerDbType.Date, new DateTime(2017, 5, 26));
+        [Fact]
+        public async Task TestBindDateNull() =>
+            await TestBind<double?>(SpannerDbType.Date, null);
+        [Fact]
+        public async Task TestBindDateArray() =>
+            await TestBind(SpannerDbType.ArrayOf(SpannerDbType.Date), new DateTime?[]
+            {
+                new DateTime(2017, 5, 26),
+                null,
+                new DateTime(2017, 5, 9),
+            });
+        [Fact]
+        public async Task TestBindDateEmptyArray() =>
+            await TestBind(SpannerDbType.ArrayOf(SpannerDbType.Date), new DateTime?[]
+              {});
+        [Fact]
+        public async Task TestBindDateNullArray() =>
+            await TestBind<DateTime?[]>(SpannerDbType.ArrayOf(SpannerDbType.Date), null);
+
+        [Fact]
+        public async Task TestBindTimestamp() =>
+            await TestBind(SpannerDbType.Timestamp, new DateTime(2017, 5, 26, 15, 0, 0));
+        [Fact]
+        public async Task TestBindTimestampNull() =>
+            await TestBind<double?>(SpannerDbType.Timestamp, null);
+        [Fact]
+        public async Task TestBindTimestampArray() =>
+            await TestBind(SpannerDbType.ArrayOf(SpannerDbType.Timestamp), new DateTime?[]
+            {
+                new DateTime(2017, 5, 26, 3, 15, 0),
+                null,
+                new DateTime(2017, 5, 9, 12, 30, 0)
+            });
+        [Fact]
+        public async Task TestBindTimestampEmptyArray() =>
+            await TestBind(SpannerDbType.ArrayOf(SpannerDbType.Timestamp), new DateTime?[]
+                {});
+        [Fact]
+        public async Task TestBindTimestampNullArray() =>
+            await TestBind<DateTime?[]>(SpannerDbType.ArrayOf(SpannerDbType.Timestamp), null);
+
+        async Task TestBind<T>(SpannerDbType parameterType, T value)
+        {
+            // ReSharper disable once RedundantAssignment
+            int rowsRead = -1;
```

Comment:
simply so when debugging you can tell based on the value of rowsRead which codepath it took.